src/features/sidebar/actions.ts

```diff
@@ -0,0 +1,14 @@
+import { store } from "@/store/appStore";
+import { SidebarPanelId } from "@/features/sidebar/types";
+
+const setState = store.setState;
+
+export const setSidebarPanelId = (panelId: SidebarPanelId): void => {
+  const { controlState } = store.getState();
+  setState({
```

Comment:
Because you use `store.getState()` here, you should also use `store.setState()` to avoid the asymetry. 

```suggestion
export const setSidebarPanelId = (panelId: SidebarPanelId): void => {
  const { controlState } = store.getState();
  state.setState({
```